### ISSUE TITLE
Feature Request - search / parts download

### ISSUE THREAD

**Issue by dustymc** [ArctosDB/arctos#6806]
**Is your feature request related to a problem? Please describe.**

There are some (valid) complaints about the (overloaded, I believe) form. Let's dump the whole thing, consolidate some issues, and build whatever's needed.

**Describe what you're trying to accomplish** 

Happy parts view/download (or a few of them, or whatever).

**Describe the solution you'd like**

Tell me what's needed in some actionable way.

**Describe alternatives you've considered**

Struggle?

**Additional context**

* https://github.com/ArctosDB/arctos/issues/6763 is a request to filter by part attribute type
* https://github.com/ArctosDB/arctos/issues/6386 is a request to view part attribute details
* https://github.com/ArctosDB/arctos/issues/6709 is a request to make the view and download consistent
* https://github.com/ArctosDB/arctos/issues/6363 is (a bit indirectly) a request for "internal" part identifiers

**Priority**

I don't use the thing, but seems pretty critical from here....

**Comment by Jegelewicz** [comment]
It may be that we need more than one THING. But here are a few "requirements"

1. include GUID
2. include all "part" fields: part name, part count, condition, disposition, remark, part ID - given the name of the tool - I think this makes great sense!
3. include part attribute fields: attribute type, attribute value, attribute units, attribute determiner, attribute date, attribute method, attribute remark - https://github.com/ArctosDB/arctos/issues/6801#issuecomment-1749396877
4. include part container and container stack 

Suggestions:

People use this for different reasons.
1. Looking at part attributes: maybe the "part view/ download" results could be split even further to one row per attribute (repeating all of the part information to get me "part attribute view/download"? 
2. Looking to find stuff: maybe the part location in "part view/ download" results could be spread out into columns?

The loan stuff must be there for a reason now, but I am not sure what that reason is, so it could be that there is a third "subform" that would make sense to do whatever that is doing now.

SO, I use the parts table view and see a row for each part in the record results

GUID | part name | part count | condition | disposition | remark | part ID | part barcode | PLP (JSON) | part attributes (JSON) 
 -- | -- | -- | -- | -- | -- | -- | -- | -- | --

but if I am interested in the attributes, from this view I can select a button and see a row for each attribute for all of the parts listed in the previous table

GUID | part name | part count | condition | disposition | remark | part ID | part barcode | part attribute type | part attribute value | part attribute units | part attribute determiner | part attribute date | part attribute method | part attribute remark
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- 

OR

select a different button and see a row for each part with the parsed out container hierarchy

GUID | part name | part count | condition | disposition | remark | part ID | part barcode | CIDn | Label_n
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- 

Is that about as clear as mud - and probably impossible to do?

**Comment by dustymc** [comment]
One advantage of multiple THINGS (views on the one form or multiple forms, one or multiple links - none of that really matters, I think) is that we could not pay the cost of loans and attributes every time the question involves containers, etc. (Details will matter, but 3x more rows being possible in simpler views is probably wildly too conservative a guess.) If those data are necessary together then that's just a cost we'll have to pay, if not maybe the extra capacity, simplicity, and room for growth would be better.  

>People use this for different reasons.

I would really like to hear from them - I'm not sure this is worth doing unless there's an actual and actionable use case (or maybe trend of them, one specialized thing is probably a job for reports or writeSQL).  Eg you've got JSON in one of your examples - anyone who can deal with that probably doesn't have much use for this, all part info is already cached and readily accessible as JSON in the main search UI. (Or maybe this providing a partID is enough reason to exist? IDK...)

**Comment by Jegelewicz** [comment]
Use cases

1. You have the one that @jebrad asked for help with - and while a report worked for him, it is a sort of hidden way to do things that would be solved with `the "part view/ download" results could be split even further to one row per attribute (repeating all of the part information to get me "part attribute view/download"`, especially if I could then filter the attributes to get just the ones I wanted.
2. I could use this when I need to remove a bunch of part attributes (for reasons - probably because I screwed something up). I just get the attribute rows I need and with a bit of smashing put them in a part attribute unloader.
3. In general, I think the part attributes thing will come in handy for a lot of question-answering, even if it means linking the data to another worksheet that has all of the "record" information (identification, event, identifiers).
4. I need to go find a bunch of parts for a loan would be resolved with `select a different button and see a row for each part with the parsed out container hierarchy` which I could just hand off to someone to go and pick.

What else?

**Comment by dustymc** [comment]
The report requested in https://github.com/ArctosDB/arctos/issues/6760 ended up having a lot of stuff that I don't think would be appropriate here, are you suggesting that wasn't necessary or ???

>bit of smashing 

You can use writesql (do the smashing up front).

**Comment by Jegelewicz** [comment]
> are you suggesting that wasn't necessary

No - but the basic ask at the beginning would have been answered with a part attribute detail.

> You can use writesql (do the smashing up front).

Perhaps, but not everyone has that capability and some human review is also a good thing (at least sometimes).

Also, I think this would help fulfill the "I want to download all my stuff" request. Letting a collection get all of their parts with associated attributes covers the parts section of the record? I assume that some collections will be too large for this, but for many smaller collections it might be an option. If nobody else thinks this is needed, why do we have what we have now? It has been requested, I just think that the current form that is trying to do everything is not making anyone happy and is probably too resource intensive.

Do we have any stats on how often this form gets used and even better, who is using it? That can help us target people who might be interested.

**Comment by jebrad** [comment]
> The report requested in #6760 ended up having a lot of stuff that I don't think would be appropriate here, 

Yeah, my report request ended up including pretty much everything I thought we might need when sending a request to our GRC people, I tried to err on the side of more fields because I didn't want to have to come back and ask you to add a couple more fields.

**Comment by dustymc** [comment]
For the sake of clarity:

* Modifying a report is trivial, it's yours and we can do with it whatever you want
* Modifying a shared tool requires communicating with The Community, which is often strangely difficult.

**Comment by dustymc** [comment]
closing for lack of interest, specific requests via new issues are always welcome
